includes/admin/reporting/class-file-downloads-logs-list-table.php

```diff
@@ -210,6 +168,76 @@ function get_logs( $log_query = array() ) {
 		return $logs_data;
 	}
 
+	/**
+	 * Retrieves the array of files associated with the log.
+	 *
+	 * This method contains a lot of logic to ensure backwards compatibility with how
+	 * the `edd_log_file_download_download_files` filter was formatted in EDD 2.x.
+	 *
+	 * @since 3.0
+	 *
+	 * @param File_Download_Log $log
+	 * @param int               $customer_id
+	 *
+	 * @return array
+	 */
+	protected function get_log_files( File_Download_Log $log, $customer_id ) {
+		$customer = ! empty( $customer_id ) ? edd_get_customer( $customer_id ) : false;
+
+		/*
+		 * Get the files associated with this download and store them in a property to prevent
+		 * multiple queries for the same download.
+		 * This is needed for backwards compatibility in the `edd_log_file_download_download_files` filter.
+		 */
+		if ( ! array_key_exists( $log->product_id, $this->queried_files ) ) {
+			$files = get_post_meta( $log->product_id, 'edd_download_files', true );
+			$this->queried_files[ $log->product_id ] = $files;
+		} else {
+			$files = $this->queried_files[ $log->product_id ];
+		}
+
+		/*
+		 * User info is needed for backwards compatibility in the `edd_log_file_download_download_files` filter.
+		 */
+		$user = ! empty( $customer->user_id ) ? get_userdata( $customer->user_id ) : false;
+
+		$user_info = ! empty( $user )
+			? array(
+				'id'    => $user->ID,
+				'email' => $user->user_email,
+				'name'  => $user->display_name,
+			)
+			: array();
+
+		/*
+		 * Build up an array of meta.
+		 */
+		$meta = edd_get_file_download_log_meta( $log->id );
+
+		// These meta keys no longer exist, but we need to create them for use in the filter.
+		$backwards_compat_meta = array(
+			'_edd_log_user_info'  => $user_info,
```

Comment:
Comparing a var_dump  from 2.11 with this PR, I think there also needs to be this line:

```php
'_edd_log_customer_id' => $customer->id,
```

Noting that my 2.11 log data does not include the `user_info` array:
```
array (size=6)
  '_edd_log_customer_id' => 
    array (size=1)
      0 => string '228' (length=3)
  '_edd_log_user_id' => 
    array (size=1)
      0 => string '1' (length=1)
  '_edd_log_file_id' => 
    array (size=1)
      0 => string '1' (length=1)
  '_edd_log_ip' => 
    array (size=1)
      0 => string '127.0.0.1' (length=9)
  '_edd_log_payment_id' => 
    array (size=1)
      0 => string '3463' (length=4)
  '_edd_log_price_id' => 
    array (size=1)
      0 => string '0' (length=1)
```

But if it's supposed to be there and I just don't happen to have it, that's fine, too. Other than the `customer_id`, this PR makes the `$meta` match the values in 2.x.

Reply:
I'm inclined to say we keep it anyway because I don't know why it was there and it won't do any harm to keep it. :shrug: